### PR TITLE
Chore(Paywalls): Constraints and allocations to support min/max sizes

### PR DESCRIPTION
**Pull request by JZDesign**
### Checklist
- [x] If applicable, unit tests
- [ ] If applicable, create follow-up issues for `purchases-ios` and hybrids

### Motivation

To support min/max sizes, and break up the big PR #4161

### Description

Pull files from #4161. These will be used to ensure that stacks will correctly layout children that have applied min/max sizes.

Without them the stack fails to assign the correct weights like:

image

---

> [!NOTE]
> **Low Risk**
> New internal layout utilities and tests only; no wiring into existing stack UI paths in this diff, so runtime paywall behavior is unchanged until follow-up integration.
> 
> **Overview**
> Adds **internal** paywall stack layout plumbing so children with **min/max `SizeConstraint`s** (especially `Fill`) get correct space sharing before the main stack work lands from the larger paywalls PR.
> 
> Introduces **`SizeConstraint` helpers** (`minimum`, `effectiveMaximum`, `clamp`) and **`allowsFlexDistribution`** so unconstrained `Fit` children are excluded from flex space distribution while `Fill`/`Fixed` (and `Fit` with a positive min) participate.
> 
> Adds **`allocateConstrainedFillSpace`** to split remaining main-axis space among `Fill` siblings: honor mins first, cap at maxes, redistribute leftovers, and keep mins even when the parent is too small (overflow sequentially instead of overlap).
> 
> Adds **`ConstrainedFillLayout`** plus **`arrangeConstrainedFillItems`** for horizontal/vertical stacks: measure non-fill children first, allocate fill sizes, then place with standard arrangements or `SPACE_BETWEEN` / `SPACE_AROUND` / `SPACE_EVENLY`, including an RTL fix for a single `SPACE_BETWEEN` item.
> 
> **Unit tests** cover allocation edge cases (min vs max priority, rounding) and RTL arrangement.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f5e657181ca499350a85e9de1124e279af01617b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by JZDesign** [comment]
@cursor review

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/ConstrainedFillLayout.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        }
+
+        class Vertical(
+            override val distribution: FlexDistribution,
+            val arrangement: Arrangement.Vertical,
+            val alignment: Alignment.Horizontal,
+        ) : Config {
+            override val orientation: Orientation = Orientation.Vertical
+        }
+    }
```
Comment (AlvaroBrey) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/ConstrainedFillLayout.kt]: these should be made stable to prevent unnecessary recompositions, I suggest just making `Horizontal` and `Vertical` `data class`es

**Review by AlvaroBrey — COMMENTED** [review]
Looks fine to me, just one minor comment

**Review by AlvaroBrey — APPROVED** [review]
👍
